pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -250,14 +283,23 @@ func convertToPermission(rule *rbacproto.AccessRule) *policyproto.Permission {
 }
 
 // convertToPrincipals converts a list of subjects to principals.
-func convertToPrincipals(bindings []*rbacproto.ServiceRoleBinding) []*policyproto.Principal {
+func convertToPrincipals(bindings []*rbacproto.ServiceRoleBinding) ([]*policyproto.Principal, []*policyproto.Principal) {
 	principals := make([]*policyproto.Principal, 0)
+	permissivePrincipals := make([]*policyproto.Principal, 0)
 	for _, binding := range bindings {
```

Comment:
The logic inside this for-loop seems a little redundant, could you refactor?
Some points
- the `continue` in line 294 just skip the next `for` statement in line 296, replace it to `else` would increase the readability
- line 291-293 and line 296-298 are doing the almost the same thing, could you merge them into one loop
- please also update the comment

Reply:
updated, for 2 since we check binding.mode before the loop and the logic is simple, I prefer to keep it straightforward.